routes/tasks.js

```diff
@@ -5,108 +5,285 @@ const router = express.Router();
+   res.status(400).json({error: e});
   }
 });
 
 
+
 router.get('/addnewtask/:id', async (req, res) => {
-  res.render('tasks/add-task',{taskid: req.params.id});
+  if (!req.session.user) {
+    return res.redirect("/");
+  }
+
+
+  try{
+  
+  const  id = validatetId(xss(req.params.id));
+  
+  const tasklist =  await tasklistData.getAllForAUser(req.session.user._id)
+ 
+  if(!tasklist.includes(req.params.id))
+  {
+    throw "TasklistId not exists for the user in session"
+   }
+ 
+  res.render('tasks/add-task',{taskid: id});
+  }  catch (e) {
+    console.log(e)
+    res.status(400).json({error: e});
+  }
 });
 
 router.post('/complete/:id', async (req, res) => {
+
+  if (!req.session.user) {
+    return res.redirect("/");
+  }
+ 
+
   try {
-    const completedtask = await tasksData.complete(req.params.id);
-    const tasklistid = await tasksData.gettasklistid(req.params.id)
+    const id = validatetId(xss(req.params.id));
+  
+   const  tasklistid = await tasksData.gettasklistid(id)
+       
+  const tasklistforuser =  await tasklistData.getAllForAUser(req.session.user._id)
+ 
+  if(!tasklistforuser.includes(tasklistid))
+  {
+    throw "TasklistId not exists for the user in session"
+   }
+ 
+    const completedtask = await tasksData.complete(id);
+
+
     res.redirect(`/tasks/tasksfortasklist/${tasklistid}`)
   } catch (e) {  
-    res.status(400).json({ message: e});
+    res.status(400).json({error: e});
   }
+  
 });
 
 router.post('/delete/:id', async (req, res) => {
+
+  if (!req.session.user) {
+    return res.redirect("/");
+  }
+
     try {
-      const deletedtask = await tasksData.remove(req.params.id);
-      const tasklistid = await tasksData.gettasklistid(req.params.id)
+      const id = validatetId(xss(req.params.id));
+    const  tasklistid = await tasksData.gettasklistid(id)
+       
+      const tasklistforuser =  await tasklistData.getAllForAUser(req.session.user._id)
+     
+      if(!tasklistforuser.includes(tasklistid))
+      {
+        throw "TasklistId not exists for the user in session"
+       }
+       
+      const deletedtask = await tasksData.remove(id);
+  
        res.redirect(`/tasks/tasksfortasklist/${tasklistid}`)
     } catch (e) {  
-      res.status(400).json({ message: e});
+      res.status(400).json({error: e});
     }
   });
 
 router.post('/', async (req, res) => {
+
+  if (!req.session.user) {
+    return res.redirect("/");
+  }
   const taskdetails = req.body;
-// console.log("HI" , req.body)
    try {
-    const {name, importance, deadlineDate} = taskdetails;
-   const newTask = await tasksData.create( name,  importance,  deadlineDate)
+    let {name, importance, deadlineDate} = taskdetails;
+    if(!verify.validString(xss(name))){
+      throw "Name is not valid";
+  }
+  
+  name=name.trim()
+
+  if(!verify.validDate(xss(deadlineDate)))
+  {
+    throw "Deadline date is Not a valid date";
+  }
+
+  importance  = parseInt(importance);
+
+  // if(!verify.validNumber(xss(importance)))
```

Comment:
I think we still need to verify importance as a valid number, parseInt returns 'NaN' if its not a number which isnt enough for an error check